geospaas_processing/copiers.py

```diff
@@ -42,31 +42,40 @@ def write_flag_file(type_in_flag_file, source_path, dataset, destination_filenam
             flag_file.write(string_to_write)
 
     def file_or_symlink_copy(self, source_paths, dataset):
-        """copy the file or a symlink of the file of dataset based on its stored local address in
-        the database."""
+        """copy the file or a symlink of the file/folder of dataset based on its stored local
+        address in the database."""
         for source_path in source_paths:
-            if os.path.isfile(source_path.uri):
-                destination_filename = os.path.join(
-                    self._destination_path, os.path.basename(source_path.uri))
+            destination_file_or_folder_name = os.path.join(
+                self._destination_path, os.path.basename(source_path.uri))
+            if exists(source_path.uri):
```

Comment:
I think it would be clearer to import `os.path` and use `os.path.exists()` explicitly, because "exists" is a very general term which can be applied to a lot of different subjects.

Reply:
That way we can not test it. Because .exists() is used .isfile() and .isdir() functions. And it is impossible to patch it in the tests. This way is more clearer in both code and test files

Reply:
if you want, I can use an alias such as `path_exists `instead of `exists `?!?

Reply:
No, it does not prevent you from mocking.
We already discussed this at length in the first [copy CLI pull request](https://github.com/nansencenter/django-geo-spaas-processing/pull/24), about mocking `os.path.isfile`.
Please use the same solution.

Reply:
Maybe I convey it wrong. Here this is not the case (the previous case). Look at this comment in that PR: https://github.com/nansencenter/django-geo-spaas-processing/pull/24#discussion_r527683675. Those are independant. But here, .isfile() is dependant to .exists(). So if I patch .exists() then isfile is not working properly.

Reply:
Ok, I did some testing and patching `os.path.exists()` causes a problem, but not for the reason you think.
`isfile()` does not use `exists()`, it uses `os.stat()`.
But here, what causes a problem is that `os.path.exists()` is used by the argument parser in the CLI.

So let's just leave it like that.

Reply:
However, this raises the point that for unit tests, you don't need to use the argument parsing in every test.
You need to have tests which test that the copier is correctly called when the CLI is given some arguments, and for the rest you can just test the copying functionalities by caling the copier methods directly.

The word "unit" in "unit tests" means that you test the smallest unit of code at a time.